### ISSUE TITLE
Office 365 Public CDN not working

### ISSUE THREAD

**Issue by harshdamaniahd** [SharePoint/sp-dev-fx-webparts#491]
I have a site  <tenant>sharepoint.com
In this site in "SiteAssets" Library , I have created a folder called webpart where I am hosting all files from SPFX temp folder
here are my cdn defined.
Get-SPOPublicCDNOrigins
![image](https://user-images.githubusercontent.com/11780473/39240151-a6f3cfea-48a0-11e8-8ac3-3c05c93eca05.png)

This is my cdn path : "cdnBasePath": "https://publiccdn.sharepointonline.com/<tenantname>.sharepoint.com/SiteAssets/Webpart/"

When i build and upload my webpart , I see the it is not able to get these files.
Can you pls tell me what I am missing

**Comment by waldekmastykarz** [comment]
Is there any reason why you are setting it all up manually, rather than having the SharePoint Framework package the scripts inside the .sppkg file and deploy them to the Office 365 CDN for you? This is by the way the default behavior, if you're using SharePoint Framework v1.4.1. Which version are you using?

**Comment by harshdamaniahd** [comment]
Yes I am using v1. 4.1.. No specific reason.I was trying the approach of making document library as cdn

**Comment by waldekmastykarz** [comment]
The challenge with doing it manually is, that in case the admin would turn off the Office 365 CDN, your web part wouldn't work anymore. When doing it automatically, SharePoint will automatically fallback to the document library URL when the CDN is disabled.

You could try to see if you could achieve the same result, by pointing to the document library URL rather than the CDN URL. Still, if you're deploying files to SharePoint, you could better follow the default behavior and have the SPFx toolchain include them in the package and have SharePoint deploy them automatically for you.

**Comment by harshdamaniahd** [comment]
@waldekmastykarz actually i am pointing to document library but it is not working.
https://publiccdn.sharepointonline.com/tenantname.sharepoint.com/SiteAssets/Webpart/
This is my root site

**Comment by waldekmastykarz** [comment]
@harshdamaniahd the steps outlined in my article were necessary in the past, when SharePoint didn't offer native support for deploying assets to the Office 365 CDN. Today however it does, so there is no true need to set it all up manually anymore.

Still, a manually setup CDN should work. Can you verify that your files stored in the webpart folder are accessible through the public CDN using the `publiccdn.ashx` handler as described in my article?

**Comment by harshdamaniahd** [comment]
Yes they are accessible

**Comment by harshdamaniahd** [comment]
Yes if i use https://tenantname.sharepoint.com/SiteAssets/Webpart/ as my baseurl , it loads file from the baseurl location.
It does not use cdn url .
Also i checked  in postman and tried to fetch file from cdn it gives me the error
![image](https://user-images.githubusercontent.com/11780473/39397318-00e9a8c2-4b1b-11e8-9ac9-f0e66ac4438c.png)

**Comment by waldekmastykarz** [comment]
This seems a lot like a similar issue to https://github.com/SharePoint/sp-dev-docs/issues/1731

**Comment by harshdamaniahd** [comment]
I guess there is no solution as of now, the issue is tenant specific.

**Comment by harshdamaniahd** [comment]
I my new tenant when i try to access json file from document library , i get the below error
![image](https://user-images.githubusercontent.com/11780473/40158763-d79c104c-59c3-11e8-87d3-b59c83821f18.png)


CDN is enabled
![image](https://user-images.githubusercontent.com/11780473/40158784-f6c1e55a-59c3-11e8-8546-9d6fdfca7451.png)

**Comment by hbgeorgian** [comment]
The CDN is restricted to accept requests coming from SharePoint.com only. It won't work from postman unless you tweak the origin

**Comment by harshdamaniahd** [comment]
This is working for files with extension css,txt
![image](https://user-images.githubusercontent.com/11780473/40189336-d2853c5e-5a19-11e8-98ff-cac4ff902f2e.png)


but not with json it gives error "403 FORBIDDEN"

**Comment by VesaJuvonen** [comment]
Since this seems to be a generic issue and NOT related on any sample in this repository, moving to the sp-dev-docs issue list at http://aka.ms/spdev-issues.

**Comment by VesaJuvonen** [comment]
This issue was moved to SharePoint/sp-dev-docs#2015
